### ISSUE TITLE
Int8.to_bytes_big_endian raises on negative numbers

### ISSUE THREAD

**Issue by hcarty** [andrenth/ocaml-stdint#7]
```ocaml
let buf = Bytes.make 8 '\x00' in
Int8.to_bytes_big_endian (Int8.of_int ~-1) buf 0
(* Exception: Invalid_argument "Char.chr" *)
```

**Comment by hcarty** [comment]
For what it's worth, I was able to work around this by using `EndianString.BigEndian.set_int8` from [ocplib-endian](https://github.com/OCamlPro/ocplib-endian)

**Comment by hcarty** [comment]
Looking at this a bit more, I think this may be sufficient:

```diff
diff --git a/lib/bytes_conv.ml b/lib/bytes_conv.ml
index b244f24..85293d4 100644
--- a/lib/bytes_conv.ml
+++ b/lib/bytes_conv.ml
@@ -47,7 +47,7 @@ module Make (I : IntSig) = struct
     let rec loop buffer i n =
       if i = 0 then ()
       else
-        let b = Char.chr (I.to_int (I.logand (I.of_int 0xFF) n)) in
+        let b = Char.unsafe_chr (I.to_int (I.logand (I.of_int 0xFF) n)) in
         let () = Bytes.set buffer (offset + i - 1) b in
         let n' = I.shift_right_logical n 8 in
         loop buffer (i - 1) n'
```

**Comment by andrenth** [comment]
I have applied your fix. Sorry for the ridiculous delay.

**Comment by hcarty** [comment]
Thanks!

**Comment by mwweissmann** [comment]
Looks good -- thanks!

**Comment by cfcs** [comment]
This still seems to trigger on `to_bytes_little_endian`
```ocaml
let buf = Bytes.make 8 '\x00' in
Int8.to_bytes_little_endian (Int8.of_int ~-2) buf 0
```
`Exception: Invalid_argument "Char.chr".`
ping @andrenth

**Comment by andrenth** [comment]
Analogous fix applied for little-endian in e1366f2
